In [12]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, mean_absolute_error,r2_score

In [13]:
tripdata = pd.read_csv("../../dataset/final_bike_sharing.csv")

In [14]:
tripdata.head(3)

tripduration  start_station_id start_station_name  start_lat  start_lon  \
0         15.53              3183     Exchange Place  40.716247 -74.033459   
1          9.17              3183     Exchange Place  40.716247 -74.033459   
2          5.90              3183     Exchange Place  40.716247 -74.033459   

   end_station_id end_station_name    end_lat    end_lon  bikeid  ...  \
0            3199     Newport Pkwy  40.728745 -74.032108   31929  ...   
1            3199     Newport Pkwy  40.728745 -74.032108   31845  ...   
2            3267     Morris Canal  40.712419 -74.038526   31697  ...   

  conditions        date      dist  birthyear  years_old  holiday      day  \
0      Clear  2018-01-01  1.084267       1992         26  holiday  weekday   
1      Clear  2018-01-01  1.084267       1969         49  holiday  weekday   
2      Clear  2018-01-01  0.415696       1994         24  holiday  weekday   

     month  seasons  gender  
0  January   winter    male  
1  January   winter  female  
2  January   winter    male  

[3 rows x 38 columns]

In [15]:
tripdata.columns

Index(['tripduration', 'start_station_id', 'start_station_name', 'start_lat',
       'start_lon', 'end_station_id', 'end_station_name', 'end_lat', 'end_lon',
       'bikeid', 'usertype', 'hour', 'min', 'temp', 'feelslike', 'dew',
       'humidity', 'precip', 'precipprob', 'snow', 'snowdepth', 'windspeed',
       'winddir', 'sealevelpressure', 'cloudcover', 'visibility',
       'solarradiation', 'uvindex', 'conditions', 'date', 'dist', 'birthyear',
       'years_old', 'holiday', 'day', 'month', 'seasons', 'gender'],
      dtype='object')

In [16]:
cols = ['tripduration','dist','hour','birthyear','temp','snowdepth']


In [17]:
df = tripdata[cols]

In [18]:
data = df.head(200)

In [19]:
data

tripduration      dist  hour  birthyear  temp  snowdepth
0           15.53  1.084267     2       1992 -11.7       0.00
1            9.17  1.084267    12       1969  -7.3       0.00
2            5.90  0.415696    14       1994  -5.2       0.00
3            4.17  0.240932    17       1991  -5.8       0.00
4           10.22  1.217917    22       1982  -7.2       0.00
..            ...       ...   ...        ...   ...        ...
195          6.57  0.711411    19       1985   8.3       0.98
196         12.30  0.415696    19       1992   8.3       0.98
197          6.92  1.203242    21       1982   8.8       0.85
198          3.70  0.415696     0       1992  10.6       0.66
199          3.43  0.415696    17       1980  15.0       0.09

[200 rows x 6 columns]

In [20]:
X = data.drop('tripduration', axis=1)
y = data['tripduration']
model_lr = LinearRegression(fit_intercept=True).fit(X, y)

In [21]:
print("Intercept is", model_lr.intercept_)
print("coeffients are ", model_lr.coef_)

Intercept is 336.2682709290446
coeffients are  [ 5.87188347 -0.12043737 -0.16709966  0.10261607  0.05244512]


In [22]:
y_pred = model_lr.predict(X)
rmse = mean_squared_error(y, y_pred, squared=False)
print("RMSE is of linear regression is = ", rmse)
print("R2 is of linear regression is = ", r2_score(y, y_pred))

RMSE is of linear regression is =  7.7794840221477815
R2 is of linear regression is =  0.22998510185987386


####  Python version Mini Model

1. Intercept is 336.2682709290446
2. coeffients are  [ 5.87188347 -0.12043737 -0.16709966  0.10261607  0.05244512]
3. RMSE is of linear regression is =  7.7794840221477815
4. R2 is of linear regression is =  0.22998510185987386

#### R Version Mini Model

Call:
lm(formula = tripduration ~ dist + hour + birthyear + temp + 
    snowdepth, data = data)

Residuals:
   Min     1Q Median     3Q    Max 
-9.628 -3.258 -1.285  0.981 78.697 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 336.26827  107.39851   3.131  0.00201 ** 
dist          5.87188    0.85580   6.861 8.93e-11 ***
hour         -0.12044    0.16271  -0.740  0.46006    
birthyear    -0.16710    0.05428  -3.078  0.00238 ** 
temp          0.10262    0.10590   0.969  0.33377    
snowdepth     0.05245    0.34221   0.153  0.87836    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 7.899 on 194 degrees of freedom
Multiple R-squared:   0.23,	Adjusted R-squared:  0.2101 
F-statistic: 11.59 on 5 and 194 DF,  p-value: 8.315e-10